### PR TITLE
[develop] Bindings - add runtime table cfg to prover index creation

### PR DESCRIPTION
**Pull request by dannywillems**
Explain your changes:
* Give runtime table cfgs to prover index creation. It only prepares the Caml code to receive the correct configuration. Empty values are given at the moment. Built on top of #13545, depends on https://github.com/o1-labs/proof-systems/pull/1115.
* This depends on https://github.com/o1-labs/snarkyjs/pull/1020 and https://github.com/o1-labs/snarkyjs-bindings/pull/79/. All changes related to proof-systems have been already merged.

To do:
- [ ] Check if empty arrays can be passed. It should be handled correctly on the js side.

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

**Comment by dannywillems** [comment]
!ci-build-me

**Comment by dannywillems** [comment]
!ci-build-me

**Comment by dannywillems** [comment]
!ci-build-me

**Comment by dannywillems** [comment]
!ci-build-me

**Comment by dannywillems** [comment]
!ci-build-me

**Comment by dannywillems** [comment]
The job "merges cleanly to berkeley" will be fixed in https://github.com/MinaProtocol/mina/pull/13751.

**Comment by dannywillems** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/crypto/kimchi_bindings/stubs/src/pasta_fq_plonk_index.rs [hunk 1]
```diff
@@ -39,6 +42,7 @@ impl ocaml::custom::Custom for CamlPastaFqPlonkIndex {
 pub fn caml_pasta_fq_plonk_index_create(
     gates: CamlPastaFqPlonkGateVectorPtr,
     public: ocaml::Int,
+    runtime_tables: Vec<CamlRuntimeTableCfg<CamlFq>>,
```
Comment (jspada) [inline src/lib/crypto/kimchi_bindings/stubs/src/pasta_fq_plonk_index.rs:45]: Just curious why not an Option?

File: src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_index.rs [hunk 2]
```diff
@@ -39,6 +42,7 @@ impl ocaml::custom::Custom for CamlPastaFpPlonkIndex {
 pub fn caml_pasta_fp_plonk_index_create(
     gates: CamlPastaFpPlonkGateVectorPtr,
     public: ocaml::Int,
+    runtime_tables: Vec<CamlRuntimeTableCfg<CamlFp>>,
```
Comment (jspada) [inline src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_index.rs:45]: Same

**Review by jspada — APPROVED** [review]
lgtm!

File: src/lib/crypto/kimchi_bindings/stubs/src/pasta_fq_plonk_index.rs (hunk 1 above)
Comment (dannywillems) [reply src/lib/crypto/kimchi_bindings/stubs/src/pasta_fq_plonk_index.rs:45]: No strong reason. As the empty list can represent `None`, I thought using a vec directly would be cleaner. Happy to change if you want. I am fine with both.

File: src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_index.rs (hunk 2 above)
Comment (dannywillems) [reply src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_index.rs:45]: See https://github.com/MinaProtocol/mina/pull/13549#discussion_r1282233283

**Pull request merged**